src/python/pants/backend/python/dependency_inference/parse_python_dependencies.py

```diff
@@ -47,23 +51,91 @@ class ParsedPythonDependencies:
 class ParsePythonDependenciesRequest:
     source: PythonSourceField
     interpreter_constraints: InterpreterConstraints
-    string_imports: bool
-    string_imports_min_dots: int
-    assets: bool
-    assets_min_slashes: int
+
+
+@union(in_scope_types=[])
+class PythonDependencyVisitorRequest:
+    pass
+
+
+@dataclass(frozen=True)
+class PythonDependencyVisitor:
+    digest: Digest
+    classname: str
```

Comment:
This should definitely have a comment/docstring explaining what it is. Also, I suggest using the word "spec" somewhere, since that already has a meaning for importing and its similar.

Reply:
I don't get the comment about "spec". What does that mean for importing?

Reply:
Did you mean like a command line target address spec?  This isn't that though, right?

Reply:
Module loading has ["specs"](https://docs.python.org/3/glossary.html#term-module-spec), but in this case, maybe we just split into `module` and `classname`?

Reply:
Oh, I wasn't aware of that terminology there. Live and learn!

But why split the string? We'd just have to join it back together to pass to the script via the env var.

Reply:
Maybe I'm not used to the way certain frameworks load classes, but `classname` to me sounds like just the class name, not the fully qualified module and classname.

With the comment though, it's less of a deal. I'll just mark resolved.